src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -36,19 +38,107 @@ internal static bool IsEnabled()
             //from DiagnosticListener right after the check. So some requests happening right after subscription starts
             //might not be instrumented. Similarly, when consumer unsubscribes, extra requests might be instumented
 
-            Guid loggingRequestId = Guid.NewGuid();
-            LogHttpRequest(request,  loggingRequestId);
+            Task<HttpResponseMessage> responseTask;
 
-            Task<HttpResponseMessage> responseTask = base.SendAsync(request, cancellationToken);
-
-            LogHttpResponse(responseTask, loggingRequestId);
+            //We need to know if activity events are enabled and send Request/Response events otherwise
+            if (s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.ActivityName))
+            {
+                if (Activity.Current != null)
+                {
+                    responseTask = SendInstrumentedAsync(request, cancellationToken);
+                }
+                else
+                {
+                    //null Activity.Current means that incoming request was not instrumented
+                    //and there is nothing we can do for outgoing request. 
+                    //Activity events are enabled so we don't send Request/Response events
+                    responseTask = base.SendAsync(request, cancellationToken);
+                }
+            }
+            else 
+            {
+                Guid loggingRequestId = Guid.NewGuid();
+                LogHttpRequest(request, loggingRequestId);
```

Comment:
As you note, these original logging message we would like to deprecate.   At the very least, please rename the LogHttpRequest, to 'LogHttpRequestDeprecated' and LogHttpReponse to LogHttpResponseDeprecated.  and similarly add a deprecated suffix to the RequestWriteName and ResponseWriteName symbols.

We believe the new functionality is a superset because activityies have both a ID and a beginning and  ending timestamp.   

I will start a separate discussion about deprecating DiagnosticSource events.